src/new/wallet.js

```diff
@@ -2469,6 +2464,285 @@ class HathorWallet extends EventEmitter {
+      const historyBuffer = [];
+
+      // Subscribe in websocket to the addresses
+      for (let address of addressesToSearch) {
+        wallet.subscribeAddress(address, this.conn);
+      }
+
+      while (hasMore === true) {
+        let response;
+        try {
+          response = await walletApi.getAddressHistoryForAwait(addressesToSearch, firstHash);
+        } catch (e) {
+          // We will retry the request that fails with client timeout
+          // in this request error we don't have the response because
+          // the client closed the connection
+          //
+          // I've tried to set a custom timeout error message in the axios config using timeoutErrorMessage parameter
+          // however the custom message is never used
+          // There are some error reports about it (https://github.com/axios/axios/issues/2716)
+          // Besides that, there are some problems happening in newer axios versions (https://github.com/axios/axios/issues/2710)
+          // One user that opened a PR for axios said he is checking the timeout error with the message includes condition
+          // https://github.com/axios/axios/pull/2874#discussion_r403753852
+          if (e.code === 'ECONNABORTED' && e.response === undefined && e.message.toLowerCase().includes('timeout')) {
+            // in this case we retry
+            continue;
+          }
+
+          // If the load wallet request fails with client timeout, we retry indefinitely
+          // however if we have another error, we have a limit number of retries
+          if (retryCount > LOAD_WALLET_MAX_RETRY) {
+            // Throw any error we don't want to handle here after retry limit is reached
+            throw e;
+          }
+
+          retryCount += 1;
+          await helpers.sleep(LOAD_WALLET_RETRY_SLEEP);
+          continue;
+        } finally {
+          // After each thin_wallet/address_history call we should release the event loop
+          await new Promise(resolve => { setTimeout(resolve, 0) });
+        }
+        // Reset retry count because the request succeeded
+        retryCount = 0;
+        const result = response.data;
+
+        if (result.success) {
+          hasMore = result.has_more;
+
+          // This will be an array of pages to avoid deconstructing the array with every page
+          historyBuffer.push(result.history);
+          if (hasMore) {
+            // Prepare parameters for next request
+            firstHash = result.first_hash;
+            const addrIndex = addressesToSearch.indexOf(result.first_address);
+            if (addrIndex === -1) {
+              throw new Error("Invalid address returned from the server.");
+            }
+            addressesToSearch = addressesToSearch.slice(addrIndex);
+          } else {
+            // save the history on storage since the pagination of this chunk is over
+            const historyLoadPartialUpdate = this.saveNewHistory([].concat(...historyBuffer));
```

Comment:
Not sure how optimized `[].concat(...historyBuffer)` is, to be honest, because this array might be huge.

Maybe we could try to use some `lodash` methods because they are direct and care for performance (and we already have this lib in the project). 

https://lodash.com/docs/4.17.15#flatten this should do the job, or `flattenDeep`. But as I said, I don't know how to compare the performance of both.

What do you think?